src/main/java/org/minions/devfund/yuri/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,136 @@
+package org.minions.devfund.yuri.moviedatabase;
+
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Represents a  movie database with attributes and behavior.
+ */
+public class MovieDatabase {
+    private List<Movie> movieList;
+    private List<Actor> actorList;
+
+    /**
+     * Constructor Method..
+     */
+    public MovieDatabase() {
+        movieList = new ArrayList<>();
+        actorList = new ArrayList<>();
+    }
+
+    /**
+     * Adds a movie to list.
+     *
+     * @param name   movie name.
+     * @param actors actors.
+     */
+    public void addMovie(final String name, final String[] actors) {
+        Movie movie = new Movie(name);
+        if (!existentMovie(movie.getName())) {
+            movieList.add(movie);
+            for (String actorName : actors) {
+                Actor actor = new Actor(actorName);
+                movie.getActors().add(actor);
+                actor.getMovies().add(movie);
+                if (!existentActor(actor.getName())) {
+                    actorList.add(actor);
+                }
+            }
+        }
+    }
+
+    /**
+     * Adds rating an specific movie.
+     *
+     * @param name   movie name.
+     * @param rating movie rating.
+     */
+    public void addRating(final String name, final double rating) {
+        Movie movie = getMovieByName(name);
+        movie.setRating(movie.getRating() + rating);
+    }
+
+    /**
+     * Updates the movie rating.
+     *
+     * @param name   movie name.
+     * @param rating movie rating.
+     */
+    public void updateRating(final String name, final double rating) {
+        Movie movie = getMovieByName(name);
+        movie.setRating(rating);
+    }
+
+    /**
+     * Gets the actor with best average rating.
+     *
+     * @return best actor.
+     */
+    public String getBestActor() {
+        return actorList.stream().max(Comparator.comparing(Actor::getAverage)).get().getName();
+    }
+
+    /**
+     * Gets the movie with the highest rating.
+     *
+     * @return best movie.
+     */
+    public String getBestMovie() {
+        return movieList.stream().max(Comparator.comparing(Movie::getRating)).get().getName();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)